### PR TITLE
docs: add info on import statements style

### PR DESCRIPTION
**Pull request by megies**
### What does this PR do?

Add info on how imports should be ordered in source code  +DOCS

### Why was it initiated?  Any relevant Issues?

That info was missing from style guide.

### PR Checklist
- [ ] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [ ] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] If the PR is making changes to documentation, docs pages can be built automatically.
      Just remove the space in the following string after the + sign: "+DOCS"
- [ ] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:"
      (you can also add "ALL" to just simply run all tests across all modules)
- [ ] All tests still pass.
- [ ] Any new features or fixed regressions are be covered via new tests.
- [ ] Any new or changed features have are fully documented.
- [ ] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

File: misc/docs/source/coding_style.rst [hunk 1]
```diff
@@ -46,6 +46,32 @@ manner:
 .. _SciPy: https://scipy.scipy.org/
 .. _matplotlib: http://matplotlib.org/
 
+Import statements in source code are grouped by ``__future__``, builtins, third
+party packages and finally obspy imports. Inside blocks ``from ...`` imports
+come after ``import ...`` statements:
```
Comment (ThomasLecocq) [inline misc/docs/source/coding_style.rst]: and finally in each block, they are sorted alphabetically

**Comment by ThomasLecocq** [comment]
image
if this is what you expected (the vertical grey "quote" line?), it looks good to me...

File: misc/docs/source/coding_style.rst [hunk 2]
```diff
@@ -46,6 +46,32 @@ manner:
...
+
+      import numpy as np
+      from decorator import decorator
+
+      from obspy.core import compatibility
+      from obspy.core.utcdatetime import UTCDateTime
+      from obspy.core.util import AttribDict, create_empty_data_chunk, NUMPY_VERSION
+      from obspy.core.util.base import _get_function_from_entry_point
+      from obspy.core.util.decorator import raise_if_masked, skip_if_no_data
+      from obspy.core.util.misc import (flat_not_masked_contiguous, get_window_times,
```
Comment (ThomasLecocq) [inline misc/docs/source/coding_style.rst]: a bit odd to have this violently, almost brutally, cut line, with a comma and open parenthesis... no ? 😡

**Comment by megies** [comment]
> (the vertical grey "quote" line?),

copy pasta indent, will get rid of it.. thanks :+1:

File: misc/docs/source/coding_style.rst (hunk 2 above)
Comment (megies) [reply misc/docs/source/coding_style.rst]: fixed.. thanks :+1:

**Comment by megies** [comment]
Fixed issues and force pushed

**Comment by QuLogic** [comment]
builtins -> standard library.

**Pull request merged**